File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/exploration/explorationtype/ExplorationType.scala
package cloud.benchflow.dsl.definition.configuration.terminationcriteria.exploration.explorationtype

/**
 * @author Jesper Findahl (jesper.findahl@gmail.com)
 *         created on 2017-05-22
 */
object ExplorationType extends Enumeration {
  type ExplorationType = Value
  val Fixed = Value("fixed")
  val Time = Value("time")
  val Precision = Value("precision")
  val MaxFailed = Value("max_failed")
}

File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/exploration/explorationtype/ExplorationType.java
package cloud.benchflow.dsl.definition.configuration.terminationcriteria.exploration.explorationtype;

/**
 * @author Jesper Findahl (jesper.findahl@gmail.com)
 *         created on 2017-06-23
 */
public enum ExplorationType {
  FIXED("fixed"),
  TIME("time"),
  PRECISION("precision"),
  MAX_FAILED("max_failed")
  ;

  private final String stringValue;

  ExplorationType(String stringValue) {
    this.stringValue = stringValue;
  }

  @Override
  public String toString() {
    return stringValue;
  }

}
